client/merchant-feedback-prompt/index.tsx

```diff
@@ -0,0 +1,226 @@
+								<FlexItem>
+									{ __(
+										'Are you satisfied with WooPayments?',
+										'woocommerce-payments'
+									) }
+								</FlexItem>
+								<FlexItem>
+									<Button
+										variant="link"
+										className="wcpay-merchant-feedback-prompt__action"
+										onClick={ () => {
+											recordEvent(
+												'wcpay_merchant_feedback_prompt_yes_click'
+											);
+											dismissPrompt();
+										} }
+									>
+										<Icon
+											icon={ 'thumbs-up' }
+											aria-label={ __(
+												'"Yes" icon',
+												'woocommerce-payments'
+											) }
+										/>
+										<span className="wcpay-merchant-feedback-prompt__action-label">
+											{ __(
+												'Yes',
+												'woocommerce-payments'
+											) }
+										</span>
+									</Button>
+								</FlexItem>
+								<FlexItem>
+									<Button
+										variant="link"
+										className="wcpay-merchant-feedback-prompt__action"
+										onClick={ () => {
+											recordEvent(
+												'wcpay_merchant_feedback_prompt_no_click'
+											);
+											dismissPrompt();
+										} }
+									>
+										<Icon
+											icon={ 'thumbs-down' }
+											aria-label={ __(
+												'"No" icon',
+												'woocommerce-payments'
+											) }
+										/>
+										<span className="wcpay-merchant-feedback-prompt__action-label">
+											{ __(
+												'No',
+												'woocommerce-payments'
+											) }
+										</span>
+									</Button>
+								</FlexItem>
+
+								{ /*
+									Explicit dismiss button, in place of missing `explicitDismiss` prop.
+									We need to implement manually because we use an outdated bundled SnackbarList component which doesn't have the `explicitDismiss` prop.
+									See https://github.com/WordPress/gutenberg/blob/c300edfebb48f79f6f0f6643ce04dd73303c5fcb/packages/components/src/snackbar/index.tsx#L166-L177
+								*/ }
+								<FlexItem>
+									<span
+										role="button"
+										aria-label={ __(
+											'Dismiss',
+											'woocommerce-payments'
+										) }
+										tabIndex={ 0 }
+										onClick={ () => {
+											recordEvent(
+												'wcpay_merchant_feedback_prompt_dismiss'
+											);
+											dismissPrompt();
+										} }
+										onKeyPress={ () => {
+											recordEvent(
+												'wcpay_merchant_feedback_prompt_dismiss'
+											);
+											dismissPrompt();
+										} }
+									>
+										{ /* Unicode character for "close" icon */ }
+										&#x2715;
+									</span>
+								</FlexItem>
+							</Flex>
+						),
+					},
+				] }
+			/>
+		</WCFooterPortal>
+	);
+};
+
+/**
+ * A wrapper component that conditionally renders the merchant feedback prompt.
+ *
+ * This is used to ensure the prompt is only rendered if the account is eligible for the campaign and the user has not dismissed the prompt.
+ */
+export default function MaybeShowMerchantFeedbackPrompt() {
```

Comment:
Naming nitpick – when this is imported / used it's referred to as `MerchantFeedbackPrompt`, which is confusing, since the actual prompt with no conditionals is `MerchantFeedbackPrompt`.

Not a blocker, but might be clearer to:

- Use `Maybe` naming in all client page components, so it's obvious that component will manage the visibility.
- OR
- Renamed the inner component to something other than `MerchantFeedbackPrompt`.

Reply:
Great idea. I've changed to use `Maybe` naming (and preferred named exports) when used to clarify this behaviour in 67c6ec917a47ec59ae3bbe44e69c03251dfe3471.

https://github.com/Automattic/woocommerce-payments/blob/67c6ec917a47ec59ae3bbe44e69c03251dfe3471/docs/typescript/general-conventions.md#L3-L17